### ISSUE TITLE
Let filtering on admin conditions with _in

### ISSUE THREAD

**Issue by eriskhan** [strapi/strapi#9727]
### Please describe your feature request

- [X] **I have created my request on the Product Board before I submitted this issue**
- [X] **I have looked at all the other requests on the Product Board before I submitted this issue**

### Summary

When creating new admin conditions, it would be really useful to create such with _in queries. Example:
```
    {
      displayName: 'Accept waybills: only allowed user',
      name: 'waybill-accept',
      plugin: 'admin',
      handler: user => ({'warehouse_to.allowed_users_in': user.id}),
    }
```
But at the moment happens this:
1. An allowed user opens a Content type. At that moment everything is fine and (s)he sees only entries for which (s)he is allowed.
2. But when (s)he tries to open the entry, he sees that it is forbidden. However, he is allowed! 

### Why is it needed?

The only workaround is creating a bunch of unnecessary and useless columns like: allowed_user_1, allowed_user_2, allowed_user_3. And managing them is just one big frustration.
I want 
1. Required entries to open
2. Not to reinvent the wheel by adding a bunch of unnecessary columns which obviously do not enhance the DB speed and do raise the transactions amount.

### Suggested solution(s)

Reworking the admin conditions core and letting users access entries.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request.

**Comment by Convly** [comment]
Hello, thanks for taking the time to write this issue.

First, the correct syntax to use within the condition handlers for `_in` is in fact `$in`. We're currently using sift and its operators under the hood, thus the allowed operators are the following: `$or`, `$eq`, `$ne`, `$in`, `$nin`, `$lt`, `$lte`, `$gt`, `$gte`, `$exists`, `$elemMatch`.

Next, in your situation, I suppose that both `warehouse` and `allowed_users` are relations. However, we're currently auto-populating only the first level of relations of entities by default (performances concerns). That's why your filter on a 2nd level relation isn't working as expected.

In summary, if the relations were to be populated **deeper than the first level**, your condition would look like this:

- Depth 2
  - either `user =>  ({ 'warehouse.allowed_users': user.id })`
  - or `user => ({ 'warehouse.allowed_users': { $elemMatch: user.id } })` 

- Depth 3
  - either `user =>  ({ 'warehouse.allowed_users': { id: user.id } })`
  - or `user => ({ 'warehouse.allowed_users': { $elemMatch: { id: user.id } } })`

**Comment by eriskhan** [comment]
Oh, this works totally fine for me! Thanks very much for your help!

By the way, I think it should be mentioned in docs, as they lack any info about admin conditions. In fact, that led me to the mistake I have done.

Thanks again @Convly!

**Comment by eriskhan** [comment]
I'm sorry for opening this issue again, however... that doesn't work. In fact, the records still are listed, but it isn't possible to open them. $elemmatch doesn't work, either doesn't { id: user.id }...

**Comment by Convly** [comment]
Concerning the documentation, we're currently creating a user-guide for this feature which explains the conditions in depth. It should be released soon :eyes:

As for your condition, as I stated in my previous message, **you can't create conditions on nested relations of depth 2+** (max 1).
In your scenario, it means that you can create a condition on any non-relational fields of a relation (`{ 'warehouse.name': 'foobar' }`) but not on its own relations (`{ 'warehouse.any_relational_field': 'foobar' }`).

It's a limitation we're aware of. And we'll have to see in the future if/how we can adjust this behaviour without breaking the app's performances.

**Comment by derrickmehaffy** [comment]
Documentation is located here: https://strapi.io/documentation/developer-docs/latest/setup-deployment-guides/configurations.html#role-based-access-control

sift.js has an `$in` operator so going to mark this FR as closed.
